src/core/tree.ml

```diff
@@ -142,12 +142,12 @@ module CP = struct
       (* First occurence of [i], register the slot as a point of reference. *)
       { pool with variables = IntMap.add i slot pool.variables }
 
-  (** [register_fv slot xs pool] registers a free variables constraint for the
+  (** [register_fv slot xs pool] registers a free variables constraint for the - @PROBLEM
       variables in [xs] on the slot [slot] of the [vars] array in [pool]. *)
   let register_fv : int -> int array -> t -> t = fun i vs pool ->
     { pool with fv_conds = IntMap.add i vs pool.fv_conds }
 
-  (** [constrained_nl i pool] tells whether index [i] in the RHS's environment
+  (** [constrained_nl i pool] tells whether index [i] in the RHS's environment - @PROBLEM
```

Comment:
The signature doesn't match.

Reply:
I'm not sure to understand what you mean: the signature is `int -> t -> bool` and the documentation applies it to an int `i` and a pool of type `t`, so it looks good to me!

Reply:
It's a mistake too.